src/utils/hooks/useCreateApplication.tsx

```diff
@@ -0,0 +1,129 @@
+        if (application && application.template && application.template.templateSections) {
+            setSerialNumber(application.serial as number)
+            const sections = application.template.templateSections.nodes.map((section) =>
+                section ? section.id : -1
+            )
+            console.log(`Success to create application ${application.serial}!`)
+        
+            createApplicationSections({ applicationId: application.id, templateSections: sections }, sectionMutation)
+        } else console.log('Create application failed - no sections!')
+    } else console.log('Create application failed - no data!')
+}
+  
+function createApplicationSections (payload: SectionPayload, sectionMutation: any) {
+    const { applicationId, templateSections } = payload
+    try {
+        templateSections.forEach((sectionId) => {
+            if (sectionId !== -1) {
+                sectionMutation({
+                    variables: {
+                        applicationId,
+                        templateSectionId: sectionId,
+                    }
+                })
+            }
+        })
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+function onCreateSectionCompleted ({ createApplicationSection }: CreateSectionMutation, responseMutation: any) {
+    if (
+        createApplicationSection &&
+        createApplicationSection.applicationSection &&
+        createApplicationSection.applicationSection.templateSection
+    ) {
+        const { applicationId } = createApplicationSection.applicationSection
+        const section = createApplicationSection.applicationSection.templateSection as TemplateSection
+        const elements = section.templateElementsBySectionId.nodes as TemplateElement[]
+        const questions = elements.filter(({category}) => category === 'QUESTION')
+        console.log(`Success to create application section ${section.title}`)
+
+        createApplicationResponse({ 
+            applicationId: applicationId as number, 
+            templateQuestions: questions
+        }, responseMutation)
+
+    } else console.log('Create application section failed - no data!')
+}
+
+function createApplicationResponse (payload: ResponsePayload, responseMutation: any) {
+    const { applicationId, templateQuestions } = payload
+    try {
+        templateQuestions.forEach(({id}) => {
+            responseMutation({
+                variables: {
+                    applicationId,
+                    templateElementId: id,
+                    timeCreated: new Date(Date.now())
+                }
+            })
+        })
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+function onCreateResponseCompleted ({ createApplicationResponse }: CreateResponseMutation) {
+    if (
+        createApplicationResponse &&
+        createApplicationResponse.applicationResponse &&
+        createApplicationResponse.applicationResponse.templateElement
```

Comment:
Can you use the optional chaining syntax here:
`if (createApplicationResponse?.applicationResponse?.applicationResponse.templateElement)`

(I think that's equivalent, not 100% cos I haven't really used it myself) [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Reply:
I'll see if I can do something better. One moment!

Reply:
Ok. I have added some changes to this now. Please have another look @CarlosNZ